src/test/java/space/obminyashka/items_exchange/service/AdvertisementServiceTest.java

```diff
@@ -0,0 +1,43 @@
+package space.obminyashka.items_exchange.service;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import space.obminyashka.items_exchange.dao.AdvertisementRepository;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.*;
+import static space.obminyashka.items_exchange.util.CategoryTestUtil.*;
+
+@SpringBootTest
+class AdvertisementServiceTest {
+
+    @MockBean
+    private AdvertisementRepository advertisementRepository;
+    @Autowired
+    private AdvertisementService advertisementService;
+
+    @ParameterizedTest
+    @MethodSource("getResultsOfInteraction")
```

Comment:
You can replace the whole method with a single annotation for providing boolean values. You would have only single true or false value